openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx

```diff
@@ -131,9 +149,96 @@ function ServiceMainTabContent({
     setIsEdit(false);
   };
 
+  const callServicePatchAPI = async (
+    serviceCategory: ServiceTypes,
+    id: string,
+    jsonPatch: any
+  ) => {
+    switch (serviceCategory) {
+      case 'databaseServices':
+        return await patchDatabaseDetails(id, jsonPatch);
+      case 'messagingServices':
+        return await patchTopicDetails(id, jsonPatch);
+      case 'dashboardServices':
+        return await patchDashboardDetails(id, jsonPatch);
+      case 'pipelineServices':
+        return await patchPipelineDetails(id, jsonPatch);
+      case 'mlmodelServices':
+        return await patchMlModelDetails(id, jsonPatch);
+      case 'storageServices':
+        return await patchContainerDetails(id, jsonPatch);
+      case 'searchServices':
+        return await patchSearchIndexDetails(id, jsonPatch);
+      case 'apiServices':
+        return await patchApiCollection(id, jsonPatch);
+      default:
+        throw new Error('Unsupported service category');
+    }
```

Comment:
Let's move this logic to the utils.

Reply:
Moved this function to `ServiceMainTabContentUtils.tsx`